proposals/csharp-8.0/ranges.md

```diff
@@ -133,13 +136,149 @@ var multiDimensional = list[1..2, ..]   // list[Range.Create(1, 2), Range.All]
 
 Moreover, `System.Index` should have an implicit conversion from `System.Int32`, in order not to need to overload for mixing integers and indexes over multi-dimensional signatures.
 
-## Workarounds
+## Adding Index and Range support to library types
```

Comment:
I would clean this up a bit by changing the language from proposal to spec. There's a lot of discussion about the design problems with the previous system which is not really relevant when the new spec is incorporated. In other words, I don't think the historical account is useful for the full spec.

Reply:
Right. I was chatting about this with @MadsTorgersen today. I do like the idea of making these more like formal specs as they move to completion.

Reply:
In this instance, I made a smaller set of changes to simplify the reading here.

As another PR, I'd like to rework this (and other proposals) to use more spec like language. Your comment above could be said for many of the proposals. In all cases, I'd like to stage the changes in two phases:

1. Get the technical details correct.
1. Update the language from spec to proposal.

We should discuss which order is better in general, but since this is in progress, I'd like to start with combining these two documents, and then address the language throughout the document in a separate PR.